### DISCUSSION TITLE
If `rails new` fails, created files are not removed

### DISCUSSION THREAD

**Discussion by louis-antonopoulos** [thoughtbot/suspenders#1308] [Ideas]
When running `rails new`, if an error is encountered, all the files that are created are not removed, including databases.

Related: can any known requirements (such as the Node version) be checked before starting?

```
rails new my-app-name \
 --skip_rubocop \
 --skip-test \
 -d=postgresql \
 -m=https://raw.githubusercontent.com/thoughtbot/suspenders/main/lib/install/web.rb
```

**Example error:**

```
The template [https://raw.githubusercontent.com/thoughtbot/suspenders/main/lib/install/web.rb] could not be loaded. Error: 

=== Node version unsupported ===

Suspenders requires Node >= 20.0.0
```

**Comment by stevepolitodesign** [comment]
Noting, this also applies to cases where the wrong arguments are not passed:

https://github.com/thoughtbot/suspenders/blob/3bc560ec285d21a8d7f6e83b228b33887190e683/lib/install/web.rb#L42-L66
